spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpRemoteFileTemplate.java

```diff
@@ -58,21 +76,71 @@ public T doInSession(Session<FTPFile> session) throws IOException {
 		});
 	}
 
+	/**
+	 * This particular FTP implementation is based on the {@link FTPClient#getStatus(String)}
+	 * by default, but since not all FTP servers properly implement the {@code STAT} command,
+	 * the framework internal {@link FtpRemoteFileTemplate} instances are switched to the
+	 * {@link FTPClient#listNames(String)} for only files operations.
+	 * <p> The mode can be switched with the {@link #setExistsMode(ExistsMode)} property.
+	 * <p> Any custom implementation can be done in an extension of the {@link FtpRemoteFileTemplate}.
+	 * @param path the remote file path to check.
+	 * @return true or false if remote file exists or not.
+	 */
 	@Override
 	public boolean exists(final String path) {
-		return executeWithClient(new ClientCallback<FTPClient, Boolean>() {
+		return doExecuteWithClient(new ClientCallback<FTPClient, Boolean>() {
 
 			@Override
 			public Boolean doWithClient(FTPClient client) {
 				try {
-					return client.getStatus(path) != null;
+					switch (FtpRemoteFileTemplate.this.existsMode) {
+
+						case STAT:
+							return client.getStatus(path) != null;
+
+						case NLIST:
+							String[] names = client.listNames(path);
+							return !ObjectUtils.isEmpty(names);
+
+						case NLIST_AND_DIRS:
+							return getSession().exists(path);
+
+						default:
+							throw new IllegalStateException("Unsupported 'existsMode': " +
+									FtpRemoteFileTemplate.this.existsMode);
+					}
 				}
 				catch (IOException e) {
-					throw new MessagingException("Failed to stat " + path, e);
+					throw new MessagingException("Failed to check the remote path for " + path, e);
 				}
 			}
+
 		});
 	}
 
+	/**
+	 * The {@link #exists(String)} operation mode.
+	 * @since 4.1.9
+	 */
+	public enum ExistsMode {
+
+		/**
+		 * Perform the {@code STAT} FTP command.
+		 * Default.
+		 */
+		STAT,
+
+		/**
+		 * Perform the {@code NLIST} FTP command.
+		 * Used as default internally by framework components for files only operations.
+		 */
+		NLIST,
+
+		/**
+		 * Perform the {@code NLIST} FTP command and fall back to
+		 * {@link FTPClient#changeWorkingDirectory(String)}.
+		 */
```

Comment:
"If you are only checking for a file existing, `NLST` is preferred (unless `STAT` is supported)."